### PR TITLE
SV: remember custom bins and log scale in user preference

### PR DESCRIPTION
**Pull request by kalletlak**
Frontend changes for https://github.com/cBioPortal/cbioportal/issues/6721

### CODE REVIEW

File: src/pages/studyView/StudyViewPageStore.ts [hunk 1]
```diff
@@ -2655,6 +2664,7 @@ export class StudyViewPageStore {
     @observable private _defualtChartsDimension = observable.map<ChartDimension>();
     @observable private _defaultChartsType = observable.map<ChartType>();
     @observable private _defaultVisibleChartIds: string[] = [];
+    @observable private _defualtClinicalDataBinFilterSet = observable.map<ClinicalDataBinFilter>();
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts]: spelling (defualt)

File: src/pages/studyView/StudyViewPageStore.ts [hunk 2]
```diff
@@ -2629,6 +2632,12 @@ export class StudyViewPageStore {
         this._filterMutatedGenesTableByCancerGenes = true;
         this._filterFusionGenesTableByCancerGenes = true;
         this._filterCNAGenesTableByCancerGenes = true;
+
+        this._clinicalDataBinFilterSet.keys().forEach(chartUniqueKey => {
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts]: i'm just curious why we wouldn't just clear the whole filter set here?

Comment (kalletlak) [reply src/pages/studyView/StudyViewPageStore.ts]: this is initialized when clinical attributes are fetched and is required to fetch clinical data. Other option is to reset it to default(_defualtClinicalDataBinFilterSet)

**Comment by kalletlak** [comment]
@zhx828 removed prettier related changes.
Regarding your second point, reset settings on the chart level. This should be addressed in another pr.

**Pull request merged**